atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CQLKeyValueService.java

```diff
@@ -153,8 +157,12 @@ protected void initializeConnectionPool() {
         clusterBuilder.withClusterName("atlas_cassandra_cluster_" + config.keyspace()); // for JMX metrics
         clusterBuilder.withCompression(Compression.LZ4);
 
-        if (config.ssl()) {
+        if (config.ssl().isPresent() && config.ssl().get()) {
             clusterBuilder.withSSL();
+        } else if (config.sslConfiguration().isPresent()) {
```

Comment:
...if you configure this, I think the expectation should be that it's respected, even if you have some left-over config item `ssl: true`.